### DISCUSSION TITLE
How should I handle crashes? (web)

### DISCUSSION THREAD

**Discussion by FredrikNoren** [DioxusLabs/dioxus#1381] [Q&A]
Right now, if my dioxus app crashes (using it as a web app), the UI just stops working.

I'd like to handle this with a couple of actions:
- I'd like to show an error screen to the user
- I'd like to send the crash report to sentry
- I'd like to recover if possible

I couldn't find any information about this in the docs.

**Comment by ealmloff** [comment]
Dioxus intercepts crashes on desktop with [catch_unwind](https://doc.rust-lang.org/std/panic/fn.catch_unwind.html). Unfortunately rust's `WASM` target does not support catching panics. Instead you will need to use a global [panic handler](https://doc.rust-lang.org/std/panic/fn.set_hook.html) with [web-sys](https://crates.io/crates/web-sys) to modify the page when an error occurs

**Reply by FredrikNoren** [reply]
@Demonthos So in general, there's no way to recover from panics? Best I can do is just to show an error screen and ask people to reload?

**Reply by FredrikNoren** [reply]
I tried this, but it doesn't render anything (using the same code in a button does work, so the render_error seems correct):

```rust
std::panic::set_hook(Box::new(|err| {
    let render_error = || {
        web_sys::window()?.document()?.body()?.set_inner_html("An error occured, reload the page to try again.");
        Some(())
    };
    render_error();
    console_error_panic_hook::hook(err);
}));
```

**Reply by ealmloff** [reply]
> @Demonthos So in general, there's no way to recover from panics? Best I can do is just to show an error screen and ask people to reload?

On the web renderer, yes
